### PR TITLE
Fix segfault on Sun when no XServer is available

### PR DESCRIPTION
**Pull request by zzxyz**

### CODE REVIEW

File: src/plat/sun/detect.c [hunk 1]
```diff
@@ -284,7 +284,10 @@ void detect_res(void)
 		safe_strncpy(res_str, "No X Server", MAX_STRLEN);
 
 		if (error)
+		{
 			ERR_REPORT("Problem detecting X display resolution.");
+			return;
```
Comment (woodruffw) [inline src/plat/sun/detect.c]: Instead of returning here, let's just move the `XCloseDisplay` below into the `if` block.

Comment (woodruffw) [reply src/plat/sun/detect.c]: (See `detect_res` in `plat/linux/detect.c`)

Comment (zzxyz) [reply src/plat/sun/detect.c]: Amended.

Comment (zzxyz) [reply src/plat/sun/detect.c]: Feel free to toss the pull request and make the change how you like if it's not right and that's easier.

Comment (woodruffw) [reply src/plat/sun/detect.c]: > Feel free to toss the pull request and make the change how you like if it's not right and that's easier.

It looks good to me! I'll merge it in a bit.

**Pull request merged**